### ISSUE TITLE
Data Provider should hide Pagination if Empty.

### ISSUE THREAD

**Issue by poirazis** [Budibase/budibase#7831]
There should be an option ( or default behavior ) of the Data Provider Component, to hide its Pagination section if it has no rows.

<img width="730" alt="Screenshot 2022-09-18 at 5 00 36 PM" src="https://user-images.githubusercontent.com/80512170/190910845-6b5c9301-cd71-430c-89c6-71c3ade4e89d.png">

**Comment by martkaczmarek** [comment]
You can set it.
Go to properties -> Conditions (all the way down) -> create new -> select Update setting -> Paginate -> set false (untick) -> create condition - _data provider rows count < 0_ or something like that

**Comment by poirazis** [comment]
hey @martkaczmarek and thanks for the tip !

**Comment by martkaczmarek** [comment]
Yeah, I can see now that when I tried it I can't reference the component's self values. 
You could reference a _State_ which would be updated with data provider's row count, but again - we can't do it automatically. 
We're seriously lacking:
1. Ability to reference other objects' values - either from inside, or outside;
2. Ability to run actions on page launch, on data provider refresh, on form elements click (not change) etc.

**Comment by martkaczmarek** [comment]
@poirazis I have an idea for a workaround:
Create two identical data providers, one inside the other. Apply the same filtering and data for both of them, and move your contents to the Child one. Turn off the pagination in the parent. Create the conditional pagination in the Child. In the child's condition, you can reference the parent row count.
